### PR TITLE
Add tests for data element Post and fix bug with AbandonChanges

### PR DESCRIPTION
**Pull request by ivarne**
Fix a bug where you had to add more than 1 issues (instead of at least 1 issue) in `AbandonAllChanges`. 

## Related Issue(s)
- This includes tests for  #870 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by martinothamar — APPROVED** [review]
Woops, missed this. Maybe a good candidate for making a `HasAbandonIssues` prop to keep the conditional only one place?

**Comment by ivarne** [comment]
> Maybe a good candidate for making a HasAbandonIssues prop to keep the conditional only one place?

Or just a `ProblemDetails GetAbandonResponse()` to internalise the mapping to a `ProblemDetails` instance as well?

**Comment by martinothamar** [comment]
I would not use the ASP.NET Core `ProblemDetails` model directly in there, I think that responsibility (deciding what to return in the API) should lie in the API/outer layer. Maybe an extension method or something?

The `IInstanceMutator` property could be something like `HasAbandonIssues(out IEnumerable<AbandonIssue> issues)` to access the actual data?

File: src/Altinn.App.Core/Internal/Patch/IPatchService.cs [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-using Altinn.App.Core.Models.Validation;
-using Altinn.Platform.Storage.Interface.Models;
-using Json.Patch;
-
-namespace Altinn.App.Core.Internal.Patch;
-
-/// <summary>
-/// Service for handling JsonPatches to data elements.
-/// </summary>
-public interface IPatchService
```
Comment (martinothamar) [inline src/Altinn.App.Core/Internal/Patch/IPatchService.cs:13]: This along with `InternalPatchService` doesn't feel great. Have you run a search to make sure noone is referencing this?

**Review by martinothamar — APPROVED** [review]

File: src/Altinn.App.Core/Internal/Patch/IPatchService.cs (hunk 1 above)
Comment (ivarne) [reply src/Altinn.App.Core/Internal/Patch/IPatchService.cs:13]: It's been pretty heavily changed in 8.5.0-x anyway, so using it won't be problem free.

**Pull request merged**